python/ray/_raylet.pyx

```diff
@@ -3831,8 +3831,11 @@ cdef class CoreWorker:
             c_vector[CObjectID] free_ids = ObjectRefsToVector(object_refs)
 
         with nogil:
-            check_status(CCoreWorkerProcess.GetCoreWorker().Delete(
-                free_ids, local_only))
+            status = CCoreWorkerProcess.GetCoreWorker().Delete(free_ids, local_only)
+            if status.IsIOError():
+                check_status(CRayStatus.UnexpectedSystemExit(status.ToString()))
```

Comment:
Instead of inspecting it here, can we do `if is IOError then UnexpectedSystemExit, else as-is` in `CoreWorker::Delete` ?

Reply:
Thoughts: if there's another code path calling Delete and get broken pipe, it should all be considered as UnexpectedSystemExit

Reply:
Synced offline. It is better to add the logic in the `CoreWorker::Delete` because it can cover the case for other languages as well.  The `CoreWorker::Delete` function is intended to be open to all languages to call and is not called in other ray internal code paths.